tensorrt_llm/_torch/pyexecutor/sampler.py

```
@@ -929,6 +929,9 @@ def _process_requests(self,
                 batched_strategy = strategies[0]
             else:
                 batched_strategy = None
+        else:
+            assert len(set(strategies)) == 1, "mixed sampler is not enabled"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)